appender-core/src/main/java/com/van/logging/LoggingEventCache.java

```diff
@@ -285,6 +285,11 @@ public boolean stop() {
             cacheMonitor.shutDown();
         }
 
+        if (null != instances) {
+            // Remove the stopped cache from the queue to avoid a leak
+            instances.remove(this);
```

Comment:
`instances `is a `static final` already set, so it wouldn't be `null`.

It might help to also do this in the dtor (or as close to one Java has):
```
    @Override
    protected void finalize() {
        try {
            instances.remove(this);
        } catch (IOException ex) {
            VansLogger.logger.error(
                String.format("LoggingEventCache %s cannot remove itself from instances: %s", this, ex)
            );
        }
}
```

Reply:
@bluedenim Forgive the lag in my reply (I was on vacation).  I can remove the if check if you think it is worth doing.  As for putting the logic in the `finalize()` method, that is a bit of an anti-pattern in Java.  You don't know when `finalize()` will be called -- it happens when the JVM decides it needs to perform garbage collection, which may not be any time near to when the object is no longer used by the code.  This means that in this case, it would not prevent the leak in all scenarios -- it would totally be dependent on the heap size and how much is in use.  Hooking into Log4j2's lifecycle is the better approach IMHO as it will get called when Log4j2 stops the appender.

Reply:
Definitely keep the current code. I was proposing to ALSO put it in the `finalize()`. But it's not that crucial for the reason you mentioned.

The if test is also not a big deal; it's just not necessary.

Reply:
I think it makes sense to leave it as it is, given that `finalize()` is [deprecated and is to be removed](https://openjdk.org/jeps/421) in a future JDK release.  Suggested alternatives are to use `try-with-resources` (we would need to make the cache implement `AutoCloseable`) or to use the [Cleaner API](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/ref/Cleaner.html).

Reply:
Sounds good. I will merge and do a release in the next day or 2